mmpose/datasets/datasets/top_down/topdown_panoptic_dataset.py

```diff
@@ -0,0 +1,258 @@
+        assert self.ann_info['num_joints'] == 21
+        self.ann_info['joint_weights'] = \
+            np.ones((self.ann_info['num_joints'], 1), dtype=np.float32)
+
+        self.coco = COCO(ann_file)
+        self.image_set_index = self.coco.getImgIds()
+        self.num_images = len(self.image_set_index)
+        self.id2name, self.name2id = self._get_mapping_id_name(self.coco.imgs)
+        self.dataset_name = 'panoptic'
+
+        self.db = self._get_db()
+
+        print(f'=> num_images: {self.num_images}')
+        print(f'=> load {len(self.db)} samples')
+
+    def _get_mapping_id_name(self, imgs):
+        """
+        Args:
+            imgs (dict): dict of image info.
+
+        Returns:
+            tuple: Image name & id mapping dicts.
+
+            - id2name (dict): Mapping image id to name.
+            - name2id (dict): Mapping image name to id.
+        """
+        id2name = {}
+        name2id = {}
+        for image_id, image in imgs.items():
+            file_name = image['file_name']
+            id2name[image_id] = file_name
+            name2id[file_name] = image_id
+
+        return id2name, name2id
+
+    def _get_db(self):
+        """Load dataset."""
+        gt_db = []
+        for index in self.image_set_index:
+            num_joints = self.ann_info['num_joints']
+
+            ann_ids = self.coco.getAnnIds(imgIds=index, iscrowd=False)
+            objs = self.coco.loadAnns(ann_ids)
+
+            rec = []
+            for obj in objs:
+                if max(obj['keypoints']) == 0:
+                    continue
+                joints_3d = np.zeros((num_joints, 3), dtype=np.float32)
+                joints_3d_visible = np.zeros((num_joints, 3), dtype=np.float32)
+
+                keypoints = np.array(obj['keypoints']).reshape(-1, 3)
+                joints_3d[:, :2] = keypoints[:, :2]
+                joints_3d_visible[:, :2] = np.minimum(1, keypoints[:, 2:3])
+
+                center, scale = self._xywh2cs(*obj['bbox'][:4])
+
+                image_file = os.path.join(self.img_prefix, self.id2name[index])
+                rec.append({
+                    'image_file': image_file,
+                    'center': center,
+                    'scale': scale,
+                    'rotation': 0,
+                    'joints_3d': joints_3d,
+                    'joints_3d_visible': joints_3d_visible,
+                    'dataset': self.dataset_name,
+                    'head_size': obj['head_size'],
+                    'bbox_score': 1
+                })
+            gt_db.extend(rec)
+        return gt_db
+
+    def _xywh2cs(self, x, y, w, h):
+        """This encodes bbox(x,y,w,w) into (center, scale)
+
+        Args:
+            x, y, w, h
+
+        Returns:
+            center (np.ndarray[float32](2,)): center of the bbox (x, y).
+            scale (np.ndarray[float32](2,)): scale of the bbox w & h.
+        """
+        aspect_ratio = self.ann_info['image_size'][0] / self.ann_info[
```

Comment:
here's a shorter version, don't know if u like the style

```python
aspect_ratio = np.true_divide(*self.ann_info['image_size'])
```

Reply:
The readability gets worse...

Reply:
true